tasks/libs/pipeline/data.py

```diff
@@ -120,15 +120,29 @@ def get_failed_jobs(project_name: str, pipeline_id: str) -> FailedJobs:
 ]
 
 
-def get_job_failure_context(job_log):
+def get_job_failure_context(job: ProjectJob, job_log: str):
     """
     Parses job logs (provided as a string), and returns the type of failure (infra or job) as well
     as the precise reason why the job failed.
     """
 
+    infra_job_failure_reasons = {
```

Comment:
I think you should rather store this in the FailedJob types, not to be forced to store this here as well. I guess we probably want to have only one reference for this failure list, wdyt?